src/practical_settings/GPG.tex

```diff
@@ -2,30 +2,43 @@
 
-When using PGP, there are a couple of things to take care of:
+When using GnuPG, there are a couple of things to take care of:
 \begin{itemize*}
   \item keylengths (see section \ref{section:keylengths})
   \item randomness (see section \ref{section:RNGs})
   \item preference of symmetric encryption algorithm (see section \ref{section:CipherSuites})
   \item preference of hash function (see section \ref{section:CipherSuites})
 \end{itemize*}
 
-Properly dealing with key material, passphrases and the web-of-trust is outside of the scope of this document. The GnuPG website\footnote{\url{http://www.gnupg.org/}} has a good tutorial on PGP.
-
-This \href{https://www.debian-administration.org/users/dkg/weblog/48}{Debian How-to}\footnote{\url{https://www.debian-administration.org/users/dkg/weblog/48}} is a great resource on upgrading your old PGP key as well as on safe default settings. This section is built based on the Debian How-to.
+Properly dealing with key material, passphrases and the web-of-trust is outside of the scope of this document. The GnuPG
+website\footnote{\url{http://www.gnupg.org}} has a good tutorial on GnuPG.
 
 \subsubsection{Hashing}
-Avoid SHA-1 in GnuPG. Edit \$HOME/.gnupg/gpg.conf:
+Avoid SHA-1 by preferring better hashing methods. GnuPG. Edit \$HOME/.gnupg/gpg.conf:
 
 \configfile{gpg.conf}{208-210}{Digest selection in GnuPG}
 
-Before you generate a new PGP key, make sure there is enough entropy available (see subsection \ref{subsec:RNG-linux}).
+\subsection{Key Generation}
+\gdef\currentsectionname{GPG}
+\gdef\currentsubsectionname{GnuPG}
+Because of lack of forward secrecy \ref{subsection:PFS} in OpenPGP it is preferable to use large asymmetric keys for long term
+communication protection. A RSA key of 8192 bits should provide enough confidentiality for the next 15+ years\footnote{\url{https://www.keylength.com}}.
+
+\configfile{new-key-generation.txt}{}{New key generation with GnuPG version 2.1}
+
+\configfile{params.txt}{}{Parameters for key generation with GnuPG version 2.1}
+
+The preferences parameters S9 to Z1 correspond to AES256, CAMELLIA256, AES192, CAMELLIA192, AES, CAMELLIA128, TWOFISH,
+SHA512, SHA384, SHA256, BZIP2, ZLIB and ZIP. The parameters 3DES, SHA-1 and uncompressed are set automatically by GnuPG.
+
+\subsection{ECC - Elliptic Curve Cryptography}
+Since the release of GnuPG version 2.1 end-2014\footnote{\url{https://www.gnupg.org/faq/whats-new-in-2.1.html}} ECC is supported. Older versions though are still widely used therefore ECC is not yet applicable in practice. 
```

Comment:
According to the announcement of [GnuPG version 2.1.18](https://lists.gnupg.org/pipermail/gnupg-announce/2017q1/000401.html) version 2.0 will not be maintained longer than 2017-12-31. I hope ECC will become non experimental soon.

So you suggest to remove the whole paragraph for now?